### PR TITLE
CODEOWNERs: Sort by path + Add unit test owners

### LINKED ISSUE #1195: CODEOWNERS additions and cleanup
**Describe the feature request**

Add the @vmware-tanzu/tkg-infra-x  team to codeowners to create continuity and move away from single points of failure for reviews related to networking, windows, loadbalancing.

Along the way, also ensure there is a review team for docs, as well as clean up individual ids in the list when they are redundant.

**Describe alternatives you've considered**

Add individuals.

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by jayunit100**
### What this PR does / why we need it

Add infra-x team to own the unit tests directory to help review small updates to them.

1) sort by path
2) add infra-x team as owners on the test repo (largely depended on by windows and ipv6 work)

Fixes #1195

### Describe testing done for PR

None

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Comment by mcwumbly** [comment]
> add infra-x team as owners on the test repo (largely depended on by windows and ipv6 work)

It is striking to me that this would mean we have different code owners for these paths:
```
/pkg/v1/providers @vmware-tanzu/tanzu-framework-reviewers @vmware-tanzu/tanzu-framework-lcm-reviewers @Anuj2512
/pkg/v1/providers/tests/unit @vmware-tanzu/tkg-infra-x
```

In general I'd think we'd want a single group to own a feature including it's tests, and this makes it _seem_ like that's not the case (at least on the surface).

Perhaps that's the right first step, but does this signal a problem with our directory structure? Are there ways we can think to restructure things a bit so this feels less surprising?

**Comment by yharish991** [comment]
just thinking: how about we group the paths by codeowners something like this https://github.com/vmware-tanzu/tanzu-framework/pull/1179/files

**Comment by zjs** [comment]
Some thoughts:

Adding infra-x makes sense to me.

Organizing the file _somehow_ seems important.

Our directory structure isn't very informative, so maybe sorting by directory structure isn't the best way to organize things. Grouping thematically with comments seems like a reasonable alternative.

Having a single set of reviewers for implementation and test code seems desirable. I'd just add `@vmware-tanzu/tkg-infra-x` to `/pkg/v1/providers` instead of separating out the test ownership.

File: CODEOWNERS [hunk 1]
```diff
@@ -2,19 +2,20 @@
 /addons @vmware-tanzu/tanzu-framework-reviewers @vmware-tanzu/tkg-addons-owners
 /addons/pinniped @vmware-tanzu/tanzu-framework-reviewers @vmware-tanzu/tkg-iam-owners
 /cmd/cli @vmware-tanzu/tanzu-framework-reviewers @Anuj2512
+/cmd/cli/plugin/package @vmware-tanzu/tanzu-framework-reviewers @vmware-tanzu/tkg-addons-owners
+/cmd/cli/plugin/pinniped-auth @vmware-tanzu/tanzu-framework-reviewers @vmware-tanzu/tkg-iam-owners
+/cmd/cli/plugin/secret @vmware-tanzu/tanzu-framework-reviewers @vmware-tanzu/tkg-addons-owners
 /cmd/cli/plugin/tkr @vmware-tanzu/tanzu-framework-reviewers @prkalle
 /pkg/v1/cli @vmware-tanzu/tanzu-framework-reviewers @miclettej @Anuj2512
 /pkg/v1/providers @vmware-tanzu/tanzu-framework-reviewers @vmware-tanzu/tanzu-framework-lcm-reviewers @Anuj2512
+/pkg/v1/providers/tests/unit @vmware-tanzu/tkg-infra-x
```
Comment (vuil) [inline CODEOWNERS:11]: Please add vmware-tanzu/tanzu-framework-reviewers to this as well as a fallback or PR will be be held up until someone from tkg-infra-x chimes in

Comment (vuil) [reply CODEOWNERS:11]: actually, agree with @zjs of just adding vmware-tanzu/tkg-infra-x to /providers.
There is also a related PR #1049 that is already  updating the same file. If it is alright, let me incorporate this change (with the sorting) into that instead.

**Comment by vuil** [comment]
Changes incorporated into #1049. Closing in favor of it.

**Pull request closed without merging**